src/hotspot/share/services/threadService.hpp

```diff
@@ -631,4 +631,11 @@ class JavaThreadSleepState : public JavaThreadStatusChanger {
   }
 };
 
+
```

Comment:
I'm finding the class name VMThreadSnapshot confusing, as it's used to create a snapshot of any thread in the VM, not of the "VMThread".  This space is crowded, we have a class ThreadSnapshot here already.

VMThreadSnapshot is an AllStatic class with one method for creating snapshots... How about ThreadSnapshotFactory?

Reply:
Renamed to ThreadSnapshotFactory